model/stats/db.go

```diff
@@ -217,14 +222,19 @@ func hourlyTestStatsForOldTasksPipeline(projectId string, requester string, star
 func getHourlyTestStatsPipeline(projectId string, requester string, start time.Time, end time.Time, tasks []string, lastUpdate time.Time, oldTasks bool) []bson.M {
 	var taskIdExpr string
 	var displayTaskLookupCollection string
+	var comment string
 	if oldTasks {
 		taskIdExpr = taskOldTaskIdKeyRef
 		displayTaskLookupCollection = task.OldCollection
+		comment = "Hourly Test Stats Old Pipeline"
 	} else {
 		taskIdExpr = taskIdKeyRef
 		displayTaskLookupCollection = task.Collection
+		comment = "Hourly Test Stats Pipeline"
 	}
+
 	pipeline := []bson.M{
+		{"$match": bson.M{"$comment": fmt.Sprintf("cache historical test stats: %s", comment)}},
```

Comment:
Adding a **comment** field for traceability, this will appear in the logs but won't affect performance.